spanner_orm/admin/column.py

```diff
@@ -39,8 +39,17 @@ def nullable(self) -> bool:
 
   def field_type(self) -> Type[field.FieldType]:
     for field_type in field.ALL_TYPES:
-      if self.spanner_type == field_type.ddl():
+      len = _get_str_len(self.spanner_type)
+      if self.spanner_type == field_type.ddl(len):
         return field_type
 
     raise error.SpannerError('No corresponding Type for {}'.format(
         self.spanner_type))
+
+
+def _get_str_len(spanner_type: str):
+  """Retrieves length for existing STRING field."""
+  if spanner_type.startswith('STRING(') and spanner_type.endswith(')'):
+    if spanner_type[7:-1].isnumeric():
```

Comment:
You can avoid slicing string multiple times:
```
str_len =  spanner_type[7:-1]
if str_len.isnumeric():
  return int(str_len)
```

Reply:
Fixed